DMCompiler/DM/Visitors/DMProcBuilder.cs

```diff
@@ -57,7 +57,16 @@ sealed class DMProcBuilder {
                 }
             }
 
-            ProcessBlockInner(procDefinition.Body, silenceEmptyBlockWarning : true);
+            if (procDefinition.Body.Statements.Length == 0) {
+                DMCompiler.Emit(WarningCode.EmptyProc, _proc.Location,"Empty proc detected - add an explicit \"return\" statement");
+                if (procDefinition.Body.SetStatements.Length > 0) { // We check this after because a proc with just set statements is still an empty proc
+                    ProcessBlockInner(procDefinition.Body, silenceEmptyBlockWarning : true);
+                }
+            } else {
+                ProcessBlockInner(procDefinition.Body, silenceEmptyBlockWarning : true);
+            }
+
+
```

Comment:
No reason to not just call `ProcessBlockInner()` regardless

```suggestion
            }
            
            ProcessBlockInner(procDefinition.Body, silenceEmptyBlockWarning : true);
```